File: app/src/main/java/me/imzack/app/end/view/contract/PlanDetailViewContract.java
package me.imzack.app.end.view.contract;

import me.imzack.app.end.model.bean.FormattedType;
import me.imzack.app.end.model.bean.FormattedPlan;

public interface PlanDetailViewContract extends BaseViewContract {

    void showInitialView(FormattedPlan formattedPlan, FormattedType formattedType);

    void onAppBarScrolled(float headerLayoutAlpha);

    void onAppBarScrolledToCriticalPoint(String toolbarTitle);

    void showPlanDeletionDialog(String content);

    void onPlanStatusChanged(boolean isCompleted);

    void showContentEditorDialog(String content);

    void onContentChanged(String newContent);

    void onStarStatusChanged(boolean isStarred);

    void onTypeOfPlanChanged(FormattedType formattedType);

    void showTypePickerDialog(int defaultTypeListPos);

    void showDeadlinePickerDialog(long defaultDeadline);

    void showReminderTimePickerDialog(long defaultReminderTime);

    void onDeadlineChanged(CharSequence newDeadline);

    void onReminderTimeChanged(CharSequence newReminderTime);

    void backToTop();

    void pressBack();
}

File: app/src/main/java/net/zackzhang/app/end/view/contract/PlanDetailViewContract.kt
package net.zackzhang.app.end.view.contract

import net.zackzhang.app.end.model.bean.FormattedPlan
import net.zackzhang.app.end.model.bean.FormattedType

interface PlanDetailViewContract : BaseViewContract {

    fun showInitialView(formattedPlan: FormattedPlan, formattedType: FormattedType)

    fun onAppBarScrolled(headerLayoutAlpha: Float)

    fun onAppBarScrolledToCriticalPoint(toolbarTitle: String)

    fun showPlanDeletionDialog(content: String)

    fun onPlanStatusChanged(isCompleted: Boolean)

    fun showContentEditorDialog(content: String)

    fun onContentChanged(newContent: String)

    fun onStarStatusChanged(isStarred: Boolean)

    fun onTypeOfPlanChanged(formattedType: FormattedType)

    fun showTypePickerDialog(defaultTypeListPos: Int)

    fun showDeadlinePickerDialog(defaultDeadline: Long)

    fun showReminderTimePickerDialog(defaultReminderTime: Long)

    fun onDeadlineChanged(newDeadline: CharSequence)

    fun onReminderTimeChanged(newReminderTime: CharSequence)

    fun backToTop()

    fun pressBack()
}
